modules/core/src/main/kotlin/com/github/l34130/mise/core/wsl/WslPathUtils.kt

```diff
@@ -0,0 +1,531 @@
+                e,
+            )
+        }
+    }
+
+    /**
+     * Discovers all WSL distributions that have mise installed.
+     *
+     * Process:
+     * 1. Query all installed WSL distributions via `wsl.exe --list --quiet`
+     * 2. For each distribution, run `wsl.exe -d <distro> which mise`
+     * 3. Return list of distributions with mise and its path
+     *
+     * @return List of WSL mise installations (may be empty)
+     */
+    fun discoverWslMise(): List<WslMiseInstallation> {
+        // Support both Windows and WSL environments
+        val wslExePath = resolveWslExePath()
+        if (wslExePath == null) {
+            logger.debug("wsl.exe not found, skipping WSL mise discovery")
+            return emptyList()
+        }
+
+        val installations = mutableListOf<WslMiseInstallation>()
+
+        try {
+            // Get list of all WSL distributions
+            val listResult = ExecUtil.execAndGetOutput(
+                GeneralCommandLine(wslExePath, "--list", "--quiet"),
+                10000
+            )
+
+            if (listResult.exitCode != 0) {
+                logger.info("WSL is not available or no distributions installed")
+                return emptyList()
+            }
+
+            // Parse distribution names (remove BOM and whitespace)
+            val distributions = listResult.stdoutLines
+                .map { it.replace("\uFEFF", "").trim() }
+                .filter {
+                    it.isNotBlank() &&
+                    !setOf("docker-desktop", "docker-desktop-data").contains(it.lowercase())
+                }
+
+            // Check each distribution for mise
+            for (distro in distributions) {
+                try {
+                    val whichResult = ExecUtil.execAndGetOutput(
+                        GeneralCommandLine(wslExePath, "-d", distro, "which", "mise"),
+                        10000
+                    )
+
+                    if (whichResult.exitCode == 0) {
+                        val misePath = whichResult.stdout.trim()
+                        if (misePath.isNotBlank()) {
+                            installations.add(WslMiseInstallation(distro, misePath))
+                            logger.info("Found mise in WSL distribution '$distro': $misePath")
+                        }
+                    }
+                } catch (e: Exception) {
+                    logger.debug("Failed to check mise in distribution '$distro'", e)
+                }
+            }
+        } catch (e: Exception) {
+            logger.warn("Failed to discover WSL mise installations", e)
+        }
+
+        return installations
+    }
+}
+
+/**
+ * Represents a mise installation in a WSL distribution.
+ *
+ * @property distribution The WSL distribution name (e.g., "Ubuntu", "Debian")
+ * @property path The Unix path to the mise executable (e.g., "/usr/bin/mise")
+ */
+data class WslMiseInstallation(
+    val distribution: String,
+    val path: String,
+)
```

Comment:
Can we use intellij's `WSLDistribution.executeOnWsl()` or [`WSLDistribution.patchCommandLine()`](https://github.com/JetBrains/intellij-community/blob/75b9f30ca268054783c8190e252e03e421097489/platform/platform-impl/src/com/intellij/execution/wsl/WSLDistribution.java#L229-L232)?